### ISSUE TITLE
Strip aesthetics inconsistency with facet_grid using switch="y"

### ISSUE THREAD

**Issue by touala** [tidyverse/ggplot2#3061]
I have an issue with strip border aesthetics when using `facet_grid(..., switch="y")` in both ggplot2 version `2.2.1` and `3.1.0`. In the example below, border size is different between x and y strip.

```r
df <- data.frame(x=c(1,1,1,1), y=c(1,1,1,1), grp1=c("A","A","B","B"), grp2=c("C","D","C","D"))
ggplot(df) +
	geom_point(aes(x=x,y=y)) +
	facet_grid(grp1~grp2) +
	theme(strip.background=element_rect(colour="black"))
ggplot(df) +
	geom_point(aes(x=x,y=y)) +
	facet_grid(grp1~grp2, switch="y") +
	theme(strip.background=element_rect(colour="black"))
```

The other `element_rect` parameters (fill, linetype, and color) do not display the same behavior and `facet_grid(..., switch="x")` work as expected.

```r
ggplot(df) +
	geom_point(aes(x=x,y=y)) +
	facet_grid(grp1~grp2, switch="x") +
	theme(strip.background=element_rect(colour="black"))
```

In `3.1.0`, I can correct y strip border size independently using `theme(strip.background.y=element_rect(...))` but it's not practical (see below).

```r
ggplot(df) +
	geom_point(aes(x=x,y=y)) +
	facet_grid(grp1~grp2, switch="y") +
	theme(strip.background=element_rect(colour="black")) +
	theme(strip.background.x=element_rect(size=0.5), strip.background.y=element_rect(size=0.25))
```

Also note that strip border has round outside corner with `facet_grid(..., switch="y")`, which is not consistent with other configurations.

Let me know if you have questions.

**Comment by karawoo** [comment]
Adding a reprex to show the plots: 

``` r
library("ggplot2")

df <- data.frame(x=c(1,1,1,1), y=c(1,1,1,1), grp1=c("A","A","B","B"), grp2=c("C","D","C","D"))
ggplot(df) +
  geom_point(aes(x=x,y=y)) +
  facet_grid(grp1~grp2) +
  theme(strip.background=element_rect(colour="black"))
```

![](https://i.imgur.com/wyu06GI.png)

``` r
ggplot(df) +
  geom_point(aes(x=x,y=y)) +
  facet_grid(grp1~grp2, switch="y") +
  theme(strip.background=element_rect(colour="black"))
```

![](https://i.imgur.com/7rxI9yN.png)

<sup>Created on 2019-01-09 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

The last time I saw [something like this](https://github.com/tidyverse/ggplot2/pull/2232#issuecomment-320807751) it was due to clipping being turned off, and was fixed in [this commit](https://github.com/tidyverse/ggplot2/pull/2232/commits/cf590fc8b4746cc381538405fedf0ad0c29b2ec0). I'll try to look into that when I get a chance, maybe the fix isn't getting carried through when the strip position is switched.

**Comment by touala** [comment]
Ok, I did not realize I could directly embed plots with `reprex`. I will do it next time.

Good luck fixing that.

**Comment by clauswilke** [comment]
Related? #2772

I agree this looks like a clipping issue. I had a look at the code when #2772 was filed and couldn‘t immediately figure out where clipping goes wrong. It‘s not obvious.

**Comment by karawoo** [comment]
Closed in #3068
